workerEngineReduce.py

```diff
@@ -350,59 +279,79 @@ def pick_candidate(self): #how the next best move is picked
                                     , 'lb_value': 0 #lower bound potency value
                                     , 'ub_value': 0 #upper bound potency value
                                     , 'n': n #total games played
-                                }         
-
-                    
-
+                                }
 
         logging.debug(f'best move is - {best_move} & win rate is - {potency} & lower bound win rate is - {lb_potency}')
         return moves, best_move, potency, (lb_potency, ub_potency), n
 
     def find_opponent_move(self, move):
-        try: # find the odds of the pgn opponent move in the opening stats from the API.
-            if move.uci() == 'e8g8': #Change the values for castling in Universal Chess Interface codes, can ignore
-                move_uci = 'e8h8'
-            elif move.uci() == 'e1g1':
-                move_uci = 'e1h1'
-            else:
-                move_uci = move.uci()
+        if move.uci() == 'e8g8': #Change the values for castling in Universal Chess Interface codes, can ignore
+            move_uci = 'e8h8'
+        elif move.uci() == 'e1g1':
+            move_uci = 'e1h1'
+        else:
+            move_uci = move.uci()
 
+        try: # find the odds of the pgn opponent move in the opening stats from the API.
             move_stats = next(item for item in self.stats['moves'] if item["uci"] == move_uci) #move stats is the next move in self stats moves which matches the move fed into function
         except:
-            raise Exception(f'Cannot find move {move_uci} in opening explorer API response') 

```

Comment:
This error message was broken, because `move_uci` only existed inside of the `try` block. I moved the `if-else` statement that declares it to the outside of `try` to fix it